pkg/build/graph/analysis/bc.go

```diff
@@ -28,6 +32,12 @@ bc:
 		for _, istNode := range g.SuccessorNodesByEdgeKind(bcNode, buildedges.BuildOutputEdgeKind) {
 			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
 				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+				key := MissingRequiredRegistryWarning
+				reason := missingRegistryMessage
+				if !imageStreamNode.IsFound {
```

Comment:
Why not just add the marker and continue inside of this `if`?  If the node isn't found, it won't have a `Status.DockerImageRepository` field.